lab_5_scrapper/scrapper.py

```diff
@@ -204,13 +367,74 @@ def prepare_environment(base_path: Union[pathlib.Path, str]) -> None:
     Args:
         base_path (Union[pathlib.Path, str]): Path where articles stores
     """
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+class CrawlerRecursive(Crawler):
+    """
+    The Crawler in a recursive manner.
+
+    """
+
+    def __init__(self, configuration: Config) -> None:
+        super().__init__(configuration)
+        self.start_url = 'https://www.securitylab.ru/news/547738.php' #self.config.get_seed_urls()[0]
+        self.urls = []
+        self.driver = webdriver.Edge()
+        self.texts = []
+        self.scroll_timeout = 5
+
+
+    #def find_articles(self) -> None:
+
+        #return super().find_articles()
+
+
+    def scroll(self) -> None: 
+        page_count = 0
+        domain = 'https://www.securitylab.ru'
+        url = self.start_url
+
+        while page_count < self.scroll_timeout:
+            response = requests.get(domain + url, self.config.config_dto.headers)
+            print(self.config.config_dto.headers)
+            soup = BeautifulSoup(response.text, 'lxml')
+            #future def extract()
+            links = soup.find_all('a')
+            for link in links:
+                link = link.get('href')
+                if link and link.startswith('/news/') and link.endswith('php') and link not in self.urls:
+                    url = link
+                    self.urls.append(url)
+                    self.driver.execute_script('window.scrollTo(0, document.body.scrollHeight);')
+                    break
+            sleep(self.timeout)
+            page_count += 1
+        self.driver.quit()
```

Comment:
I am not sure whether it is worth reviewing. I do not see recursive nature and I do not see ability to stop and continue from the moment when it stopped